runtime/doc/lsp.txt

```diff
@@ -1401,36 +1392,30 @@ Lua module: vim.lsp.client                                        *lsp-client*
-                               • client_id: client handle
-      • {on_attach}?           (`elem_or_list<fun(client: vim.lsp.Client, bufnr: integer)>`)
-                               Callback invoked when client attaches to a
-                               buffer.
-      • {trace}?               (`'off'|'messages'|'verbose'`, default: "off")
-                               Passed directly to the language server in the
-                               initialize request. Invalid/empty values will
-      • {flags}?               (`table`) A table with flags for the client.
-                               The current (experimental) flags are:
-                               • {allow_incremental_sync}? (`boolean`,
-                                 default: `true`) Allow using incremental sync
-                                 for buffer edits
-                               • {debounce_text_changes} (`integer`, default:
-                                 `150`) Debounce `didChange` notifications to
-                                 the server by the given number in
-                                 milliseconds. No debounce occurs if `nil`.
-                               • {exit_timeout} (`integer|false`, default:
-                                 `false`) Milliseconds to wait for server to
-                                 exit cleanly after sending the "shutdown"
-                                 request before sending kill -15. If set to
-                                 false, nvim exits immediately after sending
-                                 the "shutdown" request to the server.
       • {root_dir}?            (`string`) Directory where the LSP server will
                                base its workspaceFolders, rootUri, and
                                rootPath on initialization.
+      • {settings}?            (`lsp.LSPObject`) Map of language
+                               server-specific settings, decided by the
+                               client. Sent to the LS if requested via
+                               `workspace/configuration`. Keys are
+                               case-sensitive.
+      • {trace}?               (`'off'|'messages'|'verbose'`, default: "off")
+                               Passed directly to the language server in the
+                               initialize request. Invalid/empty values will
+      • {workspace_folders}?   (`lsp.WorkspaceFolder[]`) List of workspace
+                               folders passed to the language server. For
+                               backwards compatibility rootUri and rootPath
+                               are derived from the first workspace folder in
+                               this list. Can be `null` if the client supports
+                               workspace folders but none are configured. See
+                               `workspaceFolders` in LSP spec.
+      • {workspace_required}?  (`boolean`) (default false) Server requires a
```

Comment:
The default here also isn't correctly formatted, though it wasn't before.

Reply:
What is the syntax for that? https://github.com/neovim/neovim/blob/43f3c4a48f6cc3a4c4173591a59b8e3fcdb96759/runtime/lua/vim/lsp/client.lua#L135-L138

Reply:
Move `(default false)` and add `(default: `false`)` directly above the `@field` line.

```lua
 --- Server requires a workspace (no "single file" support). Note: Without 
 --- a workspace, cross-file features (navigation, hover) may or may not work depending on the 
 --- language server, even if the server doesn't require a workspace. 
 --- (default: `false`)
 --- @field workspace_required? boolean 
```

Reply:
thanks, fixed in https://github.com/neovim/neovim/pull/33687